src/androidMain/kotlin/com/liftric/job/queue/NetworkManager.kt

```diff
@@ -0,0 +1,55 @@
+package com.liftric.job.queue
+
+import android.annotation.SuppressLint
+import android.annotation.TargetApi
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Build
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+
+@SuppressLint("MissingPermission")
+actual class NetworkManager(context: Context) {
+    private var connectivityManager: ConnectivityManager =
+        context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+    private val connectionPublisher = MutableStateFlow(getCurrentNetworkConnection())
+
+    fun observeNetworkConnection(): Flow<NetworkState?> = connectionPublisher
+
+    private fun getCurrentNetworkConnection(): NetworkState? =
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            postAndroidMNetworkConnection(connectivityManager)
+        } else {
+            preAndroidMNetworkConnection(connectivityManager)
+        }
+
+    @TargetApi(Build.VERSION_CODES.M)
```

Comment:
Lovely to have it backwards compatible. We should internally discuss which API level makes sense for Flowify. I would rather start with Android 6 or 7 to get rid of some old chunks. But nice that you've implemented it with old versions in mind! Just leave it as is at the moment, we may remove it later on!